cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -359,7 +362,55 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
             } else {
                 throw e;
             }
+        } catch (IllegalArgumentException e) {
+            if (isPodStuck(pod) || restartContext.backOff.done()) {
+                if (canRoll(podId, 60_000, TimeUnit.MILLISECONDS, true)) {
+                    log.warn("{}: Pod {} will be force-rolled, due to error: {}", reconciliation, podName(podId), e.getCause() != null ? e.getCause().getMessage() : e.getMessage());
+                    restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+                } else {
+                    log.warn("{}: Pod {} can't be safely force-rolled; original error: ", reconciliation, podName(podId), e.getCause() != null ? e.getCause().getMessage() : e.getMessage());
+                    throw e;
+                }
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    private boolean isCrashLooping(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "CrashLoopBackOff");
+    }
+
+    private boolean isImagePullBackOff(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "ImagePullBackOff");
+    }
+
+    private boolean isCreating(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "ContainerCreating");
+    }
+
+    private boolean podWaitingBecauseOfReason(Pod pod, String reason) {
+        if (pod != null && pod.getStatus() != null) {
+            List<ContainerStatus> kafkaContainerStatus = pod.getStatus().getContainerStatuses().stream().filter(containerStatus -> containerStatus.getName().equals("kafka")).collect(Collectors.toList());
```

Comment:
Breaking the long line at `filter` and `collect` means it fits on a screen and makes it easier to see the logic.